cockatrice/src/dlg_update.cpp

```diff
@@ -108,40 +103,34 @@ void DlgUpdate::finishedUpdateCheck(bool needToUpdate, bool isCompatible, QVaria
     //If there are no available builds, then they can't auto update.
     enableUpdateButton(isCompatible);
 
-    //If there is an update, save its URL and work out its name
-    if (isCompatible) {
-        QString endUrl = (*build)["path"].toString();
-        updateUrl = API_DOWNLOAD_BASE_URL + endUrl;
-        commitHash = (*build)["sha1"].toString().left(SHORT_SHA1_HASH_LENGTH);
-        commitDate = (*build)["created"].toString().remove(DATE_LENGTH, MAX_DATE_LENGTH);
-    }
-
     //Give the user the appropriate message
-    if (needToUpdate) {
-        if (isCompatible) {
-
-            QMessageBox::StandardButton reply;
-            reply = QMessageBox::question(this, "Update Available",
-                                          "A new build (commit " + commitHash + ") from " + commitDate +
-                                          " is available. Download?",
-                                          QMessageBox::Yes | QMessageBox::No);
-            if (reply == QMessageBox::Yes)
-                downloadUpdate();
-        }
-        else
-        {
-            QMessageBox::information(this, tr("Cockatrice Update"),
-                                     tr("Your version of Cockatrice is out of date, but there are no packages"
-                                                " available for your operating system. You may have to use a developer build or build from source"
-                                                " yourself. Please visit the download page."));
-        }
-    }
-    else {
+    if (!needToUpdate) {
         //If there's no need to update, tell them that. However we still allow them to run the
         //downloader themselves if there's a compatible build
         QMessageBox::information(this, tr("Cockatrice Update"), tr("Your version of Cockatrice is up to date."));
     }
 
+    if (isCompatible) {
+        //If there is an update, save its URL and work out its name
+        updateUrl = release->getDownloadUrl();
+        publishDate = release->getPublishDate().toString(Qt::DefaultLocaleLongDate);
+
+        QMessageBox::StandardButton reply;
+        reply = QMessageBox::question(this, "Update Available",
```

Comment:
It seems like this is mixing UI and business logic together. An alternate approach might be to have the release updater not use any UI components but instead provide a function of `UpdateCheckResult checkForUpdates()` where `UpdateCheckResult` can be a tagged union with success+download_url, success-but-no-url-for-your-OS, you-are-up-to-date, or error-updating+the_error.

Does that make sense?

That would allow us to unit test in isolation from the UI.

Anyway I'm not going to block the PR on that change, but I'd like to hear your thoughts on it.